packages/webcrypto-ed25519-polyfill/src/secrets.ts

```diff
@@ -131,10 +131,53 @@ export function verifyPolyfill(key: CryptoKey, signature: BufferSource, data: Bu
     if (key.type !== 'public' || !key.usages.includes('verify')) {
         throw new DOMException('Unable to use this key to verify', 'InvalidAccessError');
     }
-    const publicKeyBytes = ed25519.getPublicKey(getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key));
+    const storedBytes = getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key);
     try {
-        return ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), publicKeyBytes);
+        return ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), storedBytes);
     } catch {
         return false;
     }
 }
+
+export function importKeyPolyfill(
+    format: KeyFormat,
+    keyData: BufferSource,
+    extractable: boolean,
+    keyUsages: readonly KeyUsage[]
+): CryptoKey {
+    if (keyUsages.length === 0) {
+        throw new DOMException('Usages cannot be empty when creating a key.', 'SyntaxError');
+    }
+    if (keyUsages.some(usage => PROHIBITED_KEY_USAGES.has(usage))) {
+        throw new DOMException('Unsupported key usage for an Ed25519 key.', 'SyntaxError');
+    }
+
+    const bytes = bufferSourceToUint8Array(keyData);
+
+    switch (format) {
+        case 'raw': {
```

Comment:
You have to validate that the usages includes _only_ `'verify'`. Anything else should throw `Unsupported key usage for a Ed25519 key`